backend/cmm_builtins.ml

```diff
@@ -60,6 +60,12 @@ let int_of_value arg dbg = Cop (Creinterpret_cast Int_of_value, [arg], dbg)
 
 let value_of_int arg dbg = Cop (Creinterpret_cast Value_of_int, [arg], dbg)
 
+let shift32 op arg count dbg =
+  assert (size_int = 8);
+  let mask = 32 - 1 in
+  let count = Cop (Cand, [count; Cconst_int (mask, dbg)], dbg) in
```

Comment:
Should we avoid the `Cand` if
`count` is a known value?

Reply:
I have just push a commit that adds some optimizations of constants, using existing `Cmm_helpers`. What do you think? Tested manually on a few small examples of int64 and int32.

Reply:
That's what I had in mind. The functions in
`Cmm_helpers` should optimize further if
the count is `0`. By looking at them, it seems
to happen for `lsr_int` and `asr_int`, but not
for `lsl_int`. Should we add it (in another PR)?